src/main/java/com/starkbank/utils/PaymentRequestDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.starkbank.utils;
+
+import com.google.gson.*;
+import com.starkbank.*;
+
+import java.lang.reflect.Type;
+
+public class PaymentRequestDeserializer implements JsonDeserializer<PaymentRequest> {
+
+    @Override
+    public PaymentRequest deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext ctw) throws JsonParseException {
+
+        JsonElement resourceElement = json.getAsJsonObject().get("payment");
+        json.getAsJsonObject().remove("payment");
+        PaymentRequest request = new Gson().fromJson(json, PaymentRequest.class);
+        Resource resource = null;
+        switch (request.type) {
+            case "transfer":
+                resource = new Gson().fromJson(resourceElement, Transfer.class);
+                break;
+            case "transaction":
+                resource = new Gson().fromJson(resourceElement, Transaction.class);
+                break;
+            case "boleto-payment":
+                resource = new Gson().fromJson(resourceElement, BoletoPayment.class);
+                break;
+            case "utility-payment":
+                resource = new Gson().fromJson(resourceElement, UtilityPayment.class);
+                break;
+            default:
+                throw new JsonParseException("Invalid resource type.");
```

Comment:
Acho melhor deixar como hashmap em vez de dar erro se não for reconhecido pra não quebrar o SDK de alguém que esteja numa versão antiga depois que lançarmos um tipo de pagamento novo.